pkg/block/metadata/meta.go

```diff
@@ -59,6 +59,10 @@ type Thanos struct {
 
 	// Source is a real upload source of the block.
 	Source SourceType `json:"source"`
+
+	// List of segment files (in chunks directory), in sorted order.
+	// Not used if nil/empty.
```

Comment:
Maybe let's just mention `optional`. Also, I would be somehow careful with empty. Empty / nil cases should be probably distinct. Block with no segments ... is kind of valid block, no? (:  But in worst case we do LIST call so not that bad.

```suggestion
	// List of segment files (in chunks directory), in sorted order. Optional. 
	// If nil, it means that that creator did not provide this information.
```

Reply:
Shipper doesn’t upload blocks with no segment files.

Reply:
> Shipper doesn’t upload blocks with no segment files.

My point was that block with no segments should not appear in Thanos. Also no segments = no chunks = no samples. Such block has no usable data. 

I will keep "Optional" part from your suggestion.